rhombus/data/equality.rhm

```diff
@@ -0,0 +1,44 @@
+#lang rhombus/and_meta
+
+
+export:
+  equality
+
+
+import:
+  lib("racket/base.rkt"):
+    meta
+    expose: #{generate-temporaries}
+  lib("racket/base.rkt").#{equal-hash-code}
+  lib("racket/unsafe/ops.rkt").#{unsafe-fx+/wraparound}
+  lib("racket/unsafe/ops.rkt").#{unsafe-fx*/wraparound}
+
+
+class_clause.macro 'equality: $key_expr; ...':
+  def [key_id, ...] = #{generate-temporaries}([key_expr, ...])
+  'internal PrivateForEquals
+   implements Equatable
+
+   private method $key_id(): $key_expr
+   ...
+
+   override method equals(other):
+     (other is_a PrivateForEquals)
+       $$('&& (this . $key_id() == (other -: PrivateForEquals) . $key_id())') ...
```

Comment:
Ok if I change `$$` to `$&` (so this code would need to be updated)?

I think `$&` connects better to a splicing `&` in repetitions. That is, `$&` is a combination of an escape and a splice.

Reply:
Yup, fine by me. I like `$&` better too.

Reply:
And now it's just `$`. Unlike S-expressions, a group or a list of terms doesn't correspond to a term, so `$` can just infer splicing.

Reply:
Oh nice, that's even better